### ISSUE TITLE
One  question about loss function

### ISSUE THREAD

**Issue by biyuruofeng** [open-mmlab/mmdetection#4111]
I noticed that the bbox regression of the original faster-RCNN loss function had p* in front of Lreg, but I don't see it in the mmdetection code. Would you like to answer if negative samples have been filtered out already before the return to loss function, and if so, where is the code?

```
class SmoothL1Loss(nn.Module):
    """Smooth L1 loss.

    Args:
        beta (float, optional): The threshold in the piecewise function.
            Defaults to 1.0.
        reduction (str, optional): The method to reduce the loss.
            Options are "none", "mean" and "sum". Defaults to "mean".
        loss_weight (float, optional): The weight of loss.
    """

    def __init__(self, beta=1.0, reduction='mean', loss_weight=1.0):
        super(SmoothL1Loss, self).__init__()
        self.beta = beta
        self.reduction = reduction
        self.loss_weight = loss_weight

    def forward(self,
                pred,
                target,
                weight=None,
                avg_factor=None,
                reduction_override=None,
                **kwargs):
        """Forward function.

        Args:
            pred (torch.Tensor): The prediction.
            target (torch.Tensor): The learning target of the prediction.
            weight (torch.Tensor, optional): The weight of loss for each
                prediction. Defaults to None.
            avg_factor (int, optional): Average factor that is used to average
                the loss. Defaults to None.
            reduction_override (str, optional): The reduction method used to
                override the original reduction method of the loss.
                Defaults to None.
        """
        assert reduction_override in (None, 'none', 'mean', 'sum')
        reduction = (
            reduction_override if reduction_override else self.reduction)
        loss_bbox = self.loss_weight * smooth_l1_loss(
            pred,
            target,
            weight,
            beta=self.beta,
            reduction=reduction,
            avg_factor=avg_factor,
            **kwargs)
        return loss_bbox
```

**Comment by yhcao6** [comment]
For example, Faster RCNN filter neg samples here:
https://github.com/open-mmlab/mmdetection/blob/048c41a8d00f57ed33affd0ea73c8bc6de76e98a/mmdet/models/roi_heads/bbox_heads/bbox_head.py#L161-L184
